File: Examples/CSharp/WorkingWithCalendarExceptions/AddRemoveCalendarExceptions.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithCalendarExceptions
{
    using System;

    internal class AddRemoveCalendarExceptions
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);
            
            //ExStart:AddRemoveCalendarExceptions
            //ExFor: CalendarException
            //ExFor: Calendar.Exceptions
            //ExSummary: Shows how to add/remove calendar exceptions. 
            var project = new Project(dataDir + "project_test.mpp");

            // Remove an exception
            var calendar = project.Calendars.ToList()[0];
            if (calendar.Exceptions.Count > 1)
            {
                var exception = calendar.Exceptions.ToList()[0];
                calendar.Exceptions.Remove(exception);
            }

            // Add an exception
            var calendarException = new CalendarException();
            calendarException.FromDate = new System.DateTime(2009, 1, 1);
            calendarException.ToDate = new System.DateTime(2009, 1, 3);
            calendar.Exceptions.Add(calendarException);

            // Display exceptions
            foreach (var exception in calendar.Exceptions)
            {
                Console.WriteLine("From" + exception.FromDate.ToShortDateString());
                Console.WriteLine("To" + exception.ToDate.ToShortDateString());
            }

            //ExEnd:AddRemoveCalendarExceptions
        }
    }
}

File: Examples/VisualBasic/WorkingWithCalendarExceptions/AddRemoveCalendarExceptions.vb
Imports System.IO

Namespace WorkingWithCalendarExceptions
    Public Class AddRemoveCalendarExceptions
        Public Shared Sub Run()
            ' ExStart:AddRemoveCalendarExceptions
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            Using fs As New FileStream(dataDir + "project_test.mpp", FileMode.Open)
                ' Create project instance
                Dim prj As New Project(fs)

                ' Remove an exception
                Dim cal As Aspose.Tasks.Calendar = prj.Calendars.ToList()(0)
                If cal.Exceptions.Count > 1 Then
                    Dim exc As CalendarException = cal.Exceptions.ToList()(0)
                    cal.Exceptions.Remove(exc)
                End If

                ' Add an exception
                Dim calExc As New CalendarException()
                calExc.FromDate = New System.DateTime(2009, 1, 1)
                calExc.ToDate = New System.DateTime(2009, 1, 3)
                cal.Exceptions.Add(calExc)

                ' Display exceptions
                For Each calExc1 As CalendarException In cal.Exceptions
                    Console.WriteLine("From" + calExc1.FromDate.ToShortDateString())
                    Console.WriteLine("To" + calExc1.ToDate.ToShortDateString())
                Next
            End Using
            ' ExEnd:AddRemoveCalendarExceptions
        End Sub
    End Class
End Namespace
